packages/core/src/jsx-namespace.ts

```diff
@@ -1,10 +1,8 @@
-import { SxProps } from './types'
+import { SxProp } from './types'
 
-type WithConditionalSxProps<P> = 'className' extends keyof P
-  ? P extends { className?: string }
-    ? Omit<P, keyof SxProps> & SxProps
-    : Omit<P, keyof SxProps>
-  : Omit<P, keyof SxProps>
+export type WithConditionalSxProp<P> = { className: string } extends P
```

Comment:
I like this one much more than the previous solution. I don't know why Emotion did it that way.

Reply:
Sorry, Dany. Now it's even uglier. I forgot that components can have more than one prop while writing this, and that's how I got that nice one-liner :D

Reply:
haha 😂